motor/providers/awsec2ebs/provider.go

```diff
@@ -192,9 +193,12 @@ func (p *Provider) Close() {
 	if err != nil {
 		log.Error().Err(err).Msg("unable to detach volume")
 	}
-	err = p.DeleteCreatedVolume(ctx, p.tmpInfo.scanVolumeId)
-	if err != nil {
-		log.Error().Err(err).Msg("unable to delete volume")
+	// only delete the volume if we created it, e.g., if we're scanning a snapshot
+	if p.tmpInfo.createdExtraVolume {
```

Comment:
Could we use tags for that? eg. we annotate the volume / snapshot that we created and only delete it if it has our tag. Advantage would be that we can recover from states where the execution was interrupted.

Reply:
From what I understand, the API does not support an additional filter for tags on the delete call: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteVolume.html

But perhaps @preslavgerchev or @vjeffrey know how this could work.